client/dial.go

```diff
@@ -24,10 +24,16 @@ func DefaultDialFunc(ctx context.Context, address string) (net.Conn, error) {
 // and the given TLS config will be used for encryption.
 func DialFuncWithTLS(dial DialFunc, config *tls.Config) DialFunc {
 	return func(ctx context.Context, addr string) (net.Conn, error) {
+		clonedConfig := config.Clone()
+		remoteIP, _, err := net.SplitHostPort(addr)
+		if err != nil {
+			return nil, err
+		}
+		clonedConfig.ServerName = remoteIP
```

Comment:
This makes it possible for each node to have its own certificate, however it breaks the current "simple" setup where a single certificate gets generated on a node and then shared with all the others.

What do you think of changing this to:

```golang
if clonedConfig.ServerName == "" {
		remoteIP, _, err := net.SplitHostPort(addr)
		if err != nil {
			return nil, err
		}
		clonedConfig.ServerName = remoteIP
}
```

?